### PR TITLE
mq303a: Initial implementation of the mq303a alcohol sensor

### PR DESCRIPTION
**Pull request by Jon-ICS**
The module implements support for the mq303a alcohol sensor.  It was
tested on the Grove Alcohol Sensor.

Signed-off-by: Jon Trulson <jtrulson@ics.com>
Signed-off-by: Zion Orent <zorent@ics.com>

### CODE REVIEW

File: src/mq303a/mq303a.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    MQ303A(int pin, int heaterPin);
+
+    /**
+     * MQ303A Destructor
+     */
+    ~MQ303A();
+
+    /**
+     * Get the alcohol detected from the sensor.  This value is inverted,
+     * a lower value means more alcohol detected.
```
Comment (sknepper) [inline src/mq303a/mq303a.h]: This comment seems a little misleading.  In the library, the inversion is done (1023 - mraa_aio_read(m_aio)) for the user.  In the examples, it says that a higher value from calling mq303a->value() means stronger alcohol.
Would a more appropriate comment for this function be something like:
"The value read from the analog pin is inverted.  A higher returned value means a higher amount of alcohol was detected."

Comment (Jon-ICS) [inline src/mq303a/mq303a.h]: Yes... This comment is confusing.  Yours is better.  In addition, the @return tag is also not correct for this sensor.  I can fix both if you'd like.

Comment (sknepper) [inline src/mq303a/mq303a.h]: Thanks for the confirmation!  I can make the changes when merging (I noticed the @return tag for this function as well as the next one, and made the changes locally already).  Thanks for the offer, though.  =)

Comment (Jon-ICS) [inline src/mq303a/mq303a.h]: No problem... Careful with the @ mentions though - someone named joseph hill has that name on github and is probably getting these messages :)

**Comment by sknepper** [comment]
Merged on in with the changes described.  Thanks much!

**Pull request closed without merging**